Mathlib/LinearAlgebra/Eigenspace/Matrix.lean

```diff
@@ -0,0 +1,75 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.LinearAlgebra.Eigenspace.Basic
+
+/-!
+
+# Eigenvalues, Eigenvectors and Spectrum for Matrices
+
+This file collects results about eigenvectors, eigenvalues and spectrum specific to matrices
+over a field.
+
+## Tags
+eigenspace, eigenvector, eigenvalue, eigen, spectrum, matrices, matrix
+
+-/
+
+section SpectrumDiagonal
+
+variable {R : Type*} [Field R] {n : Type*} [DecidableEq n] [Fintype n] (M : Type*)
+  [AddCommGroup M] [Module R M]
+
+open Matrix
+open Module.End
+
+/-- Basis vectors are eigenvectors of associated diagonal linear operator. -/
+lemma hasEigenvector_toLin_basis_diagonal (d : n → R) (i : n) (b : Basis n R M) :
```

Comment:
Probably `basis` doesn't need to be in the name.

Reply:
Yeah, I wasn't too sure about this. Do you think this is a case where one of these is given a prime and the other isn't? I'm not really sure of a good name for this result...